pkgs/development/libraries/boost/generic.nix

```diff
@@ -41,86 +41,57 @@ let
   # To avoid library name collisions
   layout = if taggedLayout then "tagged" else "system";
 
-  cflags = if enablePIC && enableExceptions then
-             "cflags=\"-fPIC -fexceptions\" cxxflags=-fPIC linkflags=-fPIC"
-           else if enablePIC then
-             "cflags=-fPIC cxxflags=-fPIC linkflags=-fPIC"
-           else if enableExceptions then
-             "cflags=-fexceptions"
-           else
-             "";
+  cflags = concatStringsSep " "
+    (optional (enablePIC) "-fPIC" ++
+     optional (enableExceptions) "-fexceptions");
+
+  cxxflags = optionalString (enablePIC) "-fPIC";
+
+  linkflags = optionalString (enablePIC) "-fPIC";
```

Comment:
The whole pic handling can be removed, as it is enabled by default by the compiler hardening in cc-wrapper.